app/models/ZkKafka.scala

```diff
@@ -52,7 +55,7 @@ object ZkKafka {
     val maybeData = Option(zkClient.getData.forPath(path))
 
     // log a message if we get a null result for a path read
-    maybeData.map(Option(_)).filter(_.isDefined).foreach(_ => Logger.warn(s"null data retrieved for path $path"))
+    maybeData.map(Option(_)).filter(_.isEmpty).foreach(_ => Logger.warn(s"null data retrieved for path $path"))
```

Comment:
I think the construct was a little over-complicated in the original code.  The `map(Option(_))` call looks to be redundant since it's already an `Option`, so it seems to me the whole logging line could be a simple conditional rather than monad manipulation:

`if (maybeData.isEmpty) Logger.warn(s"null data retrieved for path $path")`